sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -0,0 +1,282 @@
+import collections
+import logging
+import uuid
+import time
+from datetime import timedelta
+from typing import cast, Optional, Tuple, TYPE_CHECKING, Dict, Any
+
+try:
+    from urllib import quote_plus  # type: ignore
+except ImportError:
+    from urllib.parse import quote_plus
+
+import uamqp
+from uamqp import utils
+from uamqp.message import MessageProperties
+
+from ._common._configuration import Configuration
+from .exceptions import (
+    InvalidHandlerState,
+    ServiceBusError,
+    ServiceBusAuthorizationError,
+    _create_servicebus_exception
+)
+from ._common.utils import create_properties
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+
+_AccessToken = collections.namedtuple("AccessToken", "token expires_on")
+_LOGGER = logging.getLogger(__name__)
+
+
+def _parse_conn_str(conn_str):
+    # type: (str) -> Tuple[str, str, str, str]
+    endpoint = None
+    shared_access_key_name = None
+    shared_access_key = None
+    entity_path = None  # type: Optional[str]
+    for element in conn_str.split(";"):
+        key, _, value = element.partition("=")
+        if key.lower() == "endpoint":
+            endpoint = value.rstrip("/")
+        elif key.lower() == "hostname":
+            endpoint = value.rstrip("/")
+        elif key.lower() == "sharedaccesskeyname":
+            shared_access_key_name = value
+        elif key.lower() == "sharedaccesskey":
+            shared_access_key = value
+        elif key.lower() == "entitypath":
+            entity_path = value
+    if not all([endpoint, shared_access_key_name, shared_access_key]):
+        raise ValueError(
+            "Invalid connection string. Should be in the format: "
+            "Endpoint=sb://<FQDN>/;SharedAccessKeyName=<KeyName>;SharedAccessKey=<KeyValue>"
+        )
+    entity = cast(str, entity_path)
+    left_slash_pos = cast(str, endpoint).find("//")
+    if left_slash_pos != -1:
+        host = cast(str, endpoint)[left_slash_pos + 2:]
+    else:
+        host = str(endpoint)
+    return host, str(shared_access_key_name), str(shared_access_key), entity
+
+
+def _generate_sas_token(uri, policy, key, expiry=None):
+    # type: (str, str, str, Optional[timedelta]) -> _AccessToken
+    """Create a shared access signiture token as a string literal.
+    :returns: SAS token as string literal.
+    :rtype: str
+    """
+    if not expiry:
+        expiry = timedelta(hours=1)  # Default to 1 hour.
+
+    abs_expiry = int(time.time()) + expiry.seconds
+    encoded_uri = quote_plus(uri).encode("utf-8")  # pylint: disable=no-member
+    encoded_policy = quote_plus(policy).encode("utf-8")  # pylint: disable=no-member
+    encoded_key = key.encode("utf-8")
+
+    token = utils.create_sas_token(encoded_policy, encoded_key, encoded_uri, expiry)
+    return _AccessToken(token=token, expires_on=abs_expiry)
+
+
+class ServiceBusSharedKeyCredential(object):
```

Comment:
Nitpick: Wondering if this (+ the above sas key helper) would be better in their own file.  This is such a bikesheddy change though if you had a reason for putting it here, feel free to ignore.

Reply:
Issue created for internal structure improvement: https://github.com/Azure/azure-sdk-for-python/issues/10602